### DISCUSSION TITLE
I want integrate custom seachbox into instantsearch

### DISCUSSION THREAD

**Discussion by privacyrepo** [algolia/instantsearch#5173] [Q&A]
I want to integrate this custom searchbox into instantsearch:
https://tailwindcomponents.com/component/search-bar
```
     <div class="pt-2 relative mx-auto text-gray-600">
        <input class="border-2 border-gray-300 bg-white h-10 px-5 pr-16 rounded-lg text-sm focus:outline-none"
          type="search" name="search" placeholder="Search">
        <button type="submit" class="absolute right-0 top-0 mt-5 mr-4">
          <svg class="text-gray-600 h-4 w-4 fill-current" xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Capa_1" x="0px" y="0px"
            viewBox="0 0 56.966 56.966" style="enable-background:new 0 0 56.966 56.966;" xml:space="preserve"
            width="512px" height="512px">
            <path
              d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23  s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92  c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17  s-17-7.626-17-17S14.61,6,23.984,6z" />
          </svg>
        </button>
      </div>
```
it has svg in the custom div, which I only find out how to emded tailwindcss into searchbox classnames, like this;
```
      <SearchBox
        classNames={{
          root: "p-3 shadow-sm",
          form: "relative",
          input:
            "block w-full pl-9 pr-3 py-2 bg-white border border-slate-300 placeholder-slate-400 focus:outline-none focus:border-sky-500 focus:ring-sky-500 rounded-md focus:ring-1",
          submitIcon: "absolute top-0 left-0 bottom-0 w-6",
        }}
      />
```
but how to integrate this whole UI, do I need to write a whole new widget with useSearchBox? Thank you.
and I'm quite confused by react-instantsearch-hooks-web and react-instantsearch-hooks, and dom, stuff, also the tutorials I can find online, like youtube, stackoverflow kinda out dated since algolia updated the APIs. I was also trying to do some experiment of integrate algolia search with AstroJs, but it cannot show up in the front page, kind weired.

**Comment by Haroenv** [comment]
Hey @privacyrepo, to customise the DOM completely, you can indeed use useSearchBox with React InstantSearch hooks web (comes with components). react instantsearch hooks comes only with the hooks (eg. for React Native), React InstantSearch DOM is the previous architecture and was using HOC to allow for customisation.

If you have a sandbox of where you're stuck we may be able to help

**Reply by privacyrepo** [reply]
@Haroenv Thank you for your quick answer, and I setup the sandbox https://codesandbox.io/s/github/privacyrepo/codeboxtest
you can see I added search in the index.astro, but it just does not show up

**Reply by Haroenv** [reply]
I haven't used astro before, but it seems you need to enable client code manually: https://codesandbox.io/s/ancient-morning-upn54y?file=/src/pages/index.astro:3027-3049

**Reply by Haroenv** [reply]
if you want one at a time, you probably want a menu, but I'm not fully sure that's what you meant.
